cmd/tink-worker/internal/registry.go

```diff
@@ -64,8 +74,18 @@ func (r *RegistryConnDetails) pullImage(ctx context.Context, cli *client.Client,
 		return errors.Wrap(err, "DOCKER PULL")
 	}
 	defer out.Close()
-	if _, err := io.Copy(os.Stdout, out); err != nil {
-		return err
+	fd := json.NewDecoder(out)
+	var status *ImagePullStatus
+	for {
```

Comment:
I think we could add some units tests for this. They shouldn't be too hard to write and I think it would be better to have some internal tests, rather than no tests.